napari_arboretum/visualisation/base_plotter.py

```diff
@@ -33,23 +34,32 @@ def tracks(self) -> napari.layers.Tracks:
         """
         The napari tracks layer associated with this plotter.
         """
-        if not hasattr(self, "_tracks"):
+        if not self.has_tracks:
             raise AttributeError("No tracks set on this plotter.")
         return self._tracks
 
     @tracks.setter
     def tracks(self, track_layer: napari.layers.Tracks) -> None:
         self._tracks = track_layer
 
+    @property
+    def has_tracks(self) -> bool:
+        return hasattr(self, "_tracks")
+
     def draw_tree(self, track_id: int) -> None:
         """
         Plot the tree containing ``track_id``.
         """
         self.clear()
         root, subgraph_nodes = build_subgraph(self.tracks, track_id)
-        self.edges, self.annotations = layout_subgraph(root, subgraph_nodes)
+        self.draw_from_nodes(subgraph_nodes, track_id)
+
+    def draw_from_nodes(self, tree_nodes: List[TreeNode], track_id=None):
+        self.edges, self.annotations = layout_tree(tree_nodes)
+
+        if self.has_tracks:
+            self.update_egde_colors(update_live=False)
```

Comment:
Looks like this typo existed before

```suggestion
            self.update_edge_colors(update_live=False)
```

Reply:
😟 I shall try and add some tests to cover this in another PR!